svg.js.d.ts

```diff
@@ -1382,6 +1385,13 @@ declare module "@svgdotjs/svg.js" {
         target(): string;
     }
 
+
+    // ForeignObject.js
+    class ForeignObject extends Element {
+        constructor(node: HTMLElement, attrs?: any)
+        add(element: Dom) : ForeignObject

```

Comment:
Isnt `add` already covered in the Dom interface?

Reply:
The one in Dom is typed as

        add(element: Element, i?: number): this;

but ForeignObject accepts a Dom...

As it stands, it isn't really typeable, because the two `add` functions do different things. I can't think of a way around it apart from changing the method name in ForeignObject to something like `addHTML`.

I have to admit I don't fully understand how the source works here:

~~~ js
export default class ForeignObject extends Element {
  constructor (node, attrs = node) {
    super(nodeOrNew('foreignObject', node), attrs)
  }
}

registerMethods({
  Container: {
    foreignObject: wrapWithAttrCheck(function (width, height) {
      return this.put(new ForeignObject()).size(width, height)
    })
  }
})
~~~

The constructor takes a node, but it's called with nothing.

Reply:
Ah ok, yes, in that case foreignElement should get its own `add` typings. But dont forget the second parameter then :).

For the constructor: You CAN create an ForeignObject-Object (lol) by passing in a node. It will then use that node instead of creating a new one. Thats why there is the `nodeOrNew` call which gets passed the node and the element to create if node is not defined.

The attrs can be passed to any constructor to allow to directly set attributes on the element e.g. `new Rect({width: 10, height: 20})`